README.rst

```diff
@@ -23,22 +30,304 @@ implementation are in the accepted
+directory, and run the tests using `pytest <http://pytest.org/>`_ on this
+installed version. Note that the bulk of this command is actually defined
+in ``astropy.tests.command.AstropyTest`` - here we simply define the custom
+setuptools command.
+
+**python setup.py sdist**
+
+We redefine ``sdist`` to use the version from distutils rather than from
+setuptools, as the setuptools version requires duplication of information
+in ``MANIFEST.in``.
+
+**python setup.py build_docs**
+
+This command will automatically build the package, then run sphinx to build
+the documentation. Sphinx itself provides a custom setuptools command, which we
+expand with the following options:
+
+* ``-w``: set the return code to 1 if there are any warnings during the build
+  process.
+
+* ``-l``: completely clean previous builds, including files generated by
+  the sphinx-automodapi package (which creates API pages for different
+  functions/classes).
+
+* ``-n``: disable the intersphinx option.
+
+* ``-o``: open the documentation in a browser if a build finishes successfully.
+
+In addition, ``build_docs`` will automatically download and temporarily install
+sphinx-astropy (which is a meta-package that
+provides standardized configuration and documentation dependencies for astropy
+packages) if it isn't already installed. Temporary installation means that the
+package will be installed into an ``.eggs`` directory in the current working
+directory, and it will only be available for the duration of the call to
+``build_docs``.
+
+**python setup.py build_ext**
+
+This is also used when running ``build`` or ``install``. We add several features
+compared to the default ``build_ext`` command:
+
+* For packages with C/Cython extensions, we create a ``packagename._compiler``
+  submodule that contains information about the compilers used.
+
+* Packages that need to build C extensions using the Numpy C API, we allow
+  those packages to define the include path as ``'numpy'`` as opposed to having
+  to import Numpy and call ``get_include``. The goal is to solve the issue that
+  if one has to import Numpy to define extensions, then Numpy has to be
+  installed/available before the package is installed, which means that one
+  needs to install Numpy in a separate installation step.
+
+* We detect broken compilers and replace them with other compilers on-the-fly
+  unless the compiler is explicitly specified with the ``CC`` environment
+  variable.
+
+* If Cython is not installed, then we automatically check for generated C files
+  (which are normally present in the stable releases) and give a nice error
+  if these are not found.
+
+Version helpers
+^^^^^^^^^^^^^^^^
+
+Another piece of functionality we provide in astropy-helpers is the ability
+to generate a ``packagename.version`` file that includes functions that
+automatically set the version string for developer versions, to e.g.
+``3.2.dev22213`` so that each developer version has a unique number. To use this,
```

Comment:
This may be a bug or a feature, but people working on separate branches will share version numbers for different commits, so this uniqueness cannot be relied on on the package level.

Reply:
I've tried to clarify this.